src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseConcreteTypeAnalyzer.Collector.cs

```diff
@@ -362,6 +364,10 @@ private void GetValueTypes(List<ITypeSymbol> values, IOperation op)
                         {
                             var colOp = (ICoalesceOperation)op;
                             GetValueTypes(values, colOp.Value);
+
+                            // erase any potential nullable annotations of the left-hand value since when the value is null, it doesn't get used
+                            values[^1] = values[^1].WithNullableAnnotation(Analyzer.Utilities.Lightup.NullableAnnotation.NotAnnotated);
```

Comment:
Can `values` be still empty here? (either valid scenario or error scenario)

Reply:
Good point, I added a guard for that.